File: Examples/equirect2cubemap.cpp
#include<iostream>
#include<opencv2/opencv.hpp>
#include"../utils.hpp"
#include <opencv2/core/core.hpp>

// Creating the display window
int H = 500;
int W = 500;

std::string WINDOW_NAME{"viewer"};

int main()
{
  cv::namedWindow(WINDOW_NAME,CV_WINDOW_NORMAL);
  
  cv::Mat frame;
  cv::Mat outFrame;

  frame = cv::imread("../data/equirect_temp1.jpg");
  fisheyeImgConv mapper1;

  cv::imshow(WINDOW_NAME,frame);
  cv::waitKey(0);

  mapper1.eqrect2cubemap(frame,outFrame,256,true,true);

  cv::imshow(WINDOW_NAME,outFrame);
  cv::waitKey(0);

  mapper1.applyMap(1,frame,outFrame);
  cv::imshow(WINDOW_NAME,outFrame);
  cv::waitKey(0);
  
  return 0;
}

File: Examples/equirect2cubemap.py
#!/usr/bin/env/python
import cv2
import numpy as np
import math
import time
import sys
sys.path.append('../')
from fisheyeUtils import *


img_path = sys.argv[1]
dice = int(sys.argv[2])

equiRect = cv2.imread(img_path)

inShape = equiRect.shape[:2]
mapper = fisheyeImgConv()


##############  Uncomment any of the given block to run desired example  ######################

"""
# For single image in horiontal form
cubemap = mapper.equirect2cubemap(equiRect,side=256,modif=0,dice=dice)
cv2.namedWindow("cubemap",cv2.WINDOW_NORMAL)
# cv2.imwrite("../data/cubemap.jpg",cubemap)
cv2.imshow("cubemap",cubemap)
cv2.waitKey(0)
"""

# """
# For single image in dice form
start = time.time()
for i in range(20):
	cubemap = mapper.equirect2cubemap(equiRect,side=256,dice=dice)
print((time.time()-start)/20)
print("Input shape",equiRect.shape)
print("Output shape",cubemap.shape)
# cv2.imwrite("cubemap.jpg",cubemap)
cv2.imshow("cubemap",cubemap)
cv2.imshow("Equirect",equiRect)
cv2.waitKey(0)
# """


"""
# For multiple images and the mapping is not changing
fisheye = mapper.equirect2cubemap(equiRect,side=256,dice=True)
fisheye = mapper.applyMap(1,equiRect)
cv2.imshow("fisheye",fisheye)
cv2.waitKey(0)
"""
